src/commands/link/link.ts

```diff
@@ -322,13 +322,13 @@ export const link = async (options: OptionValues, command: BaseCommand) => {
     if (results.length === 0) {
       error(new Error(`No sites found named ${options.name}`))
     }
-    const [firstSiteData] = results
-    state.set('siteId', firstSiteData.id)
+    const matchingSiteData = results.find((site: any) => site.name === options.name) || results[0]
```

Comment:
can we avoid introducing new `any`s here?

Reply:
Created a new type `SiteInfo` and used that to avoid introducing new `any` types.